substrate/frame/balances/src/lib.rs

```diff
@@ -457,7 +396,7 @@ pub mod pallet {
 		/// resulting in an outright loss.
 		DustLost { account: T::AccountId, amount: T::Balance },
 		/// Transfer succeeded.
-		Transfer { from: T::AccountId, to: T::AccountId, amount: T::Balance },
+		Transfer { from: T::AccountId, to: T::AccountId, amount: T::Balance, memo: Option<T::Memo> },
```

Comment:
Again I would like to keep the exists event unchanged.  We can add another `TransferWithMemo` event that has all of the same values and the memo, this new event would only be emitted for transfers that have a memo.